swap/src/bitcoin/wallet.rs

```diff
@@ -163,15 +151,15 @@ impl Wallet {
     }
 
     pub async fn get_network(&self) -> bitcoin::Network {
-        self.inner.lock().await.network()
+        self.wallet.lock().await.network()
     }
 
     /// Broadcast the given transaction to the network and emit a log statement
     /// if done so successfully.
     pub async fn broadcast(&self, transaction: Transaction, kind: &str) -> Result<Txid> {
```

Comment:
I think we should just iterate over all outputs of the transaction here and subscribe to the script!